core/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java

```diff
@@ -145,6 +145,15 @@ public void testTaskCounts() {
         assertThat(response.getTasks().size(), greaterThanOrEqualTo(cluster().numDataNodes()));
     }
 
+    public void testTaskInfoToString() {
+        // Run only on data nodes
+        ListTasksResponse response = client().admin().cluster().prepareListTasks().setActions(ListTasksAction.NAME).get();
+        assertThat(response.getTasks().size(), equalTo(1));
+        String taskInfoString = response.getTasks().get(0).toString();
+        assertThat(taskInfoString, startsWith("{"));
+        assertThat(taskInfoString, endsWith("}"));
```

Comment:
Not really, if it cannot create a proper json [it returns a error message as a string](https://github.com/elastic/elasticsearch/blob/ded694fc83af615f980a031cd058e3a13c42bebe/core/src/main/java/org/elasticsearch/common/Strings.java#L876). I will add a test that check the content.